Fix handling CallbackBindings

otsch opened the pull request:
CallbackBindings are the only Bindings where you don't need to define a JSON fieldname. Also handle the Binding if the property name of the binding doesn't match a JSON fieldname.

#37

otsch committed:
Fix handling CallbackBindings

CallbackBindings are the only Bindings where you don't need to define a JSON fieldname. Also handle the Binding if the property name of the binding doesn't match a JSON fieldname.

otsch commented:
Another option would be to add a fieldname to the CallbackBinding class. But if that's required in order to work it'd still be breaking. So I chose this solution.

fetzi commented on src/ClassBindings.php:
```diff
@@ -178,4 +190,14 @@ private function kebapToCamelCase(string $input)
+    /**
+     * @return CallbackBinding[]
+     */
+    private function callbackBindings(): array
```
maybe we can move the detection of callbackBindings to the `register` method and store them in a seperate array?

otsch replied:
👍 done

otsch committed:
Register CallbackBindings separately

As CallbackBindings aren't bound to a JSON fieldname for flexibility, store them in a separate property on registration instead of filtering them from the main bindings array.

otsch committed:
Update changelog date
